src/components/kubectl/proxy.ts

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as opn from 'opn';
+
+import {createReadStream, exists, open, unlink} from 'fs';
+import {resolve} from 'path';
+
+const KUBE_DASHBOARD_URL = "http://localhost:8001/ui/";
+const TERMINAL_NAME = "Kubernetes Proxy";
+const PROXY_OUTPUT_FILE = resolve(__dirname, 'proxy.out');
+
+// The instance of the terminal running Kubectl Proxy.
+let terminal:vscode.Terminal;
+
+/**
+ * Runs `kubectl proxy` in a terminal process spawned by the extension, and opens the Kubernetes
+ * dashboard in the user's preferred browser.
+ */
+export async function proxyKubernetes () {
+    // If we've already got a terminal instance, just open the dashboard.
+    if (terminal) {
+        opn(KUBE_DASHBOARD_URL);
+        return;
+    }
+
+    let outputExists;
+
+    try {
+        outputExists = await promisedExists(PROXY_OUTPUT_FILE);
+        if (!outputExists) {
+            await promisedOpen(PROXY_OUTPUT_FILE);
+        }
+    } catch (e) {
+        console.log("Something went wrong when ensuring Kubernetes API Proxy output stream existed");
+        return;
+    }
+
+    // Read proxy's stdout as a stream.
+    const proxyIn = createReadStream(
+        PROXY_OUTPUT_FILE,
+        {encoding: 'utf8'}
+    ).on('data', readStream);
+
+    terminal = vscode.window.createTerminal(TERMINAL_NAME);
+    vscode.window.onDidCloseTerminal(onClosedTerminal);
+
+    // stdout is also written to a file via `tee`. We read this file as a stream
+    // to listen for when the server is ready.
+    terminal.sendText(`kubectl proxy | tee ${PROXY_OUTPUT_FILE}`, true);
+    terminal.show();
+}
+
+/**
+ * Called when the terminal window is closed by the user.
+ * @param proxyTerminal
+ */
+const onClosedTerminal = async (proxyTerminal:vscode.Terminal) => {
```

Comment:
The extension has traditionally been inconsistent about this but is there a particular reason for preferring `const foo = (bar) => { ... };` to `function foo(bar) { ... }`?

Reply:
I (slightly) prefer arrow syntax, as it appears cleaner to me. The use of `const` makes the function immutable, which I think is good style.